packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -288,48 +273,179 @@ class CupertinoDialogAction extends StatelessWidget {
   }
 }
 
-const double _kButtonBarHeight = 45.0;
+/// Constructs a text content section typically used in a [CupertinoAlertDialog].
+///
+/// If [title] is missing, then only [message] is added.  If [message] is
+/// missing, then only [title] is added. If both are missing, then it returns
+/// null.
+class _CupertinoAlertTitleSection extends StatelessWidget {
+  const _CupertinoAlertTitleSection({
+    Key key,
+    this.title,
+    this.message,
+    this.scrollController,
+  }) : super(key: key);
 
-// This color isn't correct. Instead, we should carve a hole in the dialog and
-// show more of the background.
-const Color _kButtonDividerColor = const Color(0xFFD5D5D5);
+  /// The (optional) title of the dialog is displayed in a large font at the top
+  /// of the dialog.
+  ///
+  /// Typically a [Text] widget.
+  final Widget title;
+
+  /// The (optional) content of the dialog is displayed in the center of the
+  /// dialog in a lighter font.
+  ///
+  /// Typically a [Text] widget.
+  final Widget message;
+
+  /// A scroll controller that can be used to control the scrolling of the message
+  /// in the dialog.
+  ///
+  /// Defaults to null, and is typically not needed, since most alert messages are short.
+  final ScrollController scrollController;
 
-class _CupertinoButtonBar extends StatelessWidget {
-  const _CupertinoButtonBar({
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> titleContentGroup = <Widget>[];
+    if (title != null) {
+      titleContentGroup.add(new Padding(
+        padding: new EdgeInsets.only(
+          left: _kEdgePadding,
+          right: _kEdgePadding,
+          bottom: message == null ? _kEdgePadding : 1.0,
+          top: _kEdgePadding,
+        ),
+        child: new DefaultTextStyle(
+          style: _kCupertinoDialogTitleStyle,
+          textAlign: TextAlign.center,
+          child: title,
+        ),
+      ));
+    }
+
+    if (message != null) {
+      titleContentGroup.add(
+        new Padding(
+          padding: new EdgeInsets.only(
+            left: _kEdgePadding,
+            right: _kEdgePadding,
+            bottom: _kEdgePadding,
+            top: title == null ? _kEdgePadding : 1.0,
+          ),
+          child: new DefaultTextStyle(
+            style: _kCupertinoDialogContentStyle,
+            textAlign: TextAlign.center,
+            child: message,
+          ),
+        ),
+      );
+    }
+
+    if (titleContentGroup.isEmpty) {
+      return null;
```

Comment:
Does this work? I suspect it might crash.

Reply:
Yes, it works.  The method returning null is private and the code that uses it does null check.  I'd rather not return null but Dart doesn't have optional types like Swift or Kotlin.  I'll comment it in the code to avoid others from questioning.